### DISCUSSION TITLE
contract as service in IFTTT

### DISCUSSION THREAD

**Discussion by sreuland** [stellar/sorobanathon#37] [General]
Hello, I am an employee of SDF and learning about Soroban contracts also. While reading up, started thinking of how these contract functions could plug into IoT or some type of `as-a-service` model and started exploring further, wanted to share outcome on a POC I tried with [IFTTT](https://ifttt.com)('If this then that') platform. I wrapped a Soroban smart contract into an IFTTT Integration Service(shim API) to be used for triggers and actions in IFTTT Applets by end users.

A [video clip](https://www.veed.io/view/ebe6248a-8139-4a6b-acc3-ab9dc8b86f0a) showing the POC in action on IFTTT, creating and running Applets which wire up Soroban contracts with other cloud services like Google Tasks and [Pushover](https://pushover.net/).

[Source code/docs](https://github.com/sreuland/ifttt_demo/) of this POC.

feedback most welcome, thanks!

**Comment by rice2000** [comment]
Thanks for sharing this, @sreuland!  It's great.  Cool to see the demo video alongside the source.
